File: src/twofivetwo/Player.java
package twofivetwo;

import static twofivetwo.HelperFunctions.input;
import static twofivetwo.HelperFunctions.print;

public class Player
{
    private String name = "";
    private double points;

    public Player(){
        this.name = input("Enter Player Name: ");
        print("Welcome ", this.name);
        this.points = 0;
    }

    public Player(String playerName) {
        this.name = playerName;
        print("Welcome ", this.name);
        this.points = 0;
    }

    public Player(Player p)
    {
        name = p.name;
        p.name = "Overwritten Name";
    }

    public String getName()
    {
        return name;
    }

    public void setName(String x)
    {
        name = x;
    }

    public double getPoints()
    {
        return points;
    }

}

File: src/twofivetwo/Player.kt
package twofivetwo

import twofivetwo.HelperFunctions.input
import twofivetwo.HelperFunctions.print

class Player {
    private var name = ""
    var points = 0.0
        private set

    constructor() {
        name = input("Enter Player Name:")
        print("Welcome $name")
        points = 0.0
    }

    constructor(playerName: String) {
        name = playerName
        print("Welcome $name")
        points = 0.0
    }

    constructor(p: Player) {
        name = p.name
        points = p.points
    }

    fun getName(): String {
        return name
    }

    fun setName(x: String) {
        name = x
    }
}